taskRequests/details/script.js

```diff
@@ -0,0 +1,241 @@
+  APPROVED: 'APPROVED',
+};
+
+let taskRequest;
+
+const taskRequestSkeleton = document.querySelector('.taskRequest__skeleton');
+const taskSkeleton = document.querySelector('.task__skeleton');
+const requestorSkeleton = document.querySelector(
+  '.requestors__container__list__skeleton',
+);
+
+const taskRequestContainer = document.getElementById('task-request-details');
+const taskContainer = document.getElementById('task-details');
+const requestorsContainer = document.getElementById('requestors-details');
+
+const taskRequestId = new URLSearchParams(window.location.search).get('id');
+history.pushState({}, '', window.location.href);
+
+let taskId;
+
+function renderTaskRequestDetails(taskRequest) {
+  taskRequestContainer.append(
+    createCustomElement({
+      tagName: 'h1',
+      textContent: `Task Request `,
+      class: 'taskRequest__title',
+      child: [
+        createCustomElement({
+          tagName: 'span',
+          class: 'taskRequest__title__subtitle',
+          textContent: `#${taskRequest.id}`,
+        }),
+      ],
+    }),
+    createCustomElement({
+      tagName: 'p',
+      textContent: 'Status: ',
+      class: 'taskRequest__status',
+      child: [
+        createCustomElement({
+          tagName: 'span',
+          textContent: taskRequest.status,
+          class: [
+            'taskRequest__status__chip',
+            `taskRequest__status__chip--${taskRequest.status.toLowerCase()}`,
+          ],
+        }),
+      ],
+    }),
+  );
+}
+
+async function renderTaskDetails(taskId) {
+  try {
+    const res = await fetch(`${API_BASE_URL}/tasks/${taskId}/details`);
+    taskSkeleton.classList.add('hidden');
+    const data = await res.json();
+
+    const { taskData } = data ?? {};
+
+    taskContainer.append(
+      createCustomElement({
+        tagName: 'h2',
+        class: 'task__title',
+        textContent: taskData?.title || 'N/A',
+      }),
+      createCustomElement({
+        tagName: 'p',
+        class: 'task_type',
+        textContent: 'Type: ',
+        child: [
+          taskData?.type
+            ? createCustomElement({
+                tagName: 'span',
+                class: [
+                  'task__type__chip',
+                  `task__type__chip--${taskData.type}`,
+                ],
+                textContent: taskData?.type,
+              })
+            : '',
+          taskData?.isNoteworthy
+            ? createCustomElement({
+                tagName: 'span',
+                class: ['task__type__chip', `task__type__chip--noteworthy`],
+                textContent: 'Note worthy',
+              })
+            : '',
+        ],
+      }),
+      createCustomElement({
+        tagName: 'p',
+        class: 'task__createdBy',
+        textContent: `Created By: `,
+        child: [
+          createCustomElement({
+            tagName: 'a',
+            href: `https://members.realdevsquad.com/${taskData.createdBy}`,
+            textContent: taskData?.createdBy || 'N/A',
```

Comment:
NIT - can we please URL as a constant here pls

Reply:
I looked into the members site, it looks like my implementation was lacking some context, so chaning it from a link to a text for now